### PR TITLE
0.2/687 arc reference with cc prefix small

### PR DESCRIPTION
**Pull request by charlesstrange2**
So, this was a bit twistyturny. I feel like my model is mildly polluted becuase ARC need CC data in the rvid. There's a lot of relationship wrangling to make that happen.

### CODE REVIEW

File: app/Family.php [hunk 1]
```diff
@@ -3,6 +3,8 @@
 namespace App;
 
 use Illuminate\Database\Eloquent\Model;
+use Log;
+use Auth;
```
Comment (katjam) [inline app/Family.php:7]: Not alphabetical but meh.

File: app/Family.php [hunk 2]
```diff
@@ -18,7 +20,9 @@ class Family extends Model
      * @var array
      */
     protected $fillable = [
-        'rvid', 'leaving_on', 'leaving_reason',
+        'leaving_on',
+        'leaving_reason',
+        'centre_sequence',
```
Comment (katjam) [inline app/Family.php:25]: nice name!

File: app/Family.php [hunk 3]
```diff
@@ -206,21 +212,48 @@ public function getEligibleChildrenCountAttribute()
...
+     * Generates and sets the components required for an RVID.
+     *
+     * @param Centre $centre
+     */
+    public function generateRVID(Centre $centre)
+    {
+        if (is_null($this->centre_sequence)) {
+            if ($centre) {
+                // Get the last family made in this centre.
+                $family = Family::where('initial_centre_id', $centre->id)
```
Comment (katjam) [inline app/Family.php:226]: Do we have soft Deletes on family - should we try to get trashed ones?

File: tests/Unit/Models/FamilyModelTest.php [hunk 4]
```diff
@@ -38,7 +40,7 @@ public function itHasAnAttributeThatCalculatesSumOfEligibleChildren()
     public function itHasAnAttributeThatReturnsNearestDueDateOrNull()
     {
         // Create Family
-        $family = factory(Family::class)->create();
+        $family = factory(Family::class)->create([]);
```
Comment (katjam) [inline tests/Unit/Models/FamilyModelTest.php:43]: I guess we don;t really need the [] but meh

File: tests/Unit/Models/FamilyModelTest.php [hunk 5]
```diff
@@ -70,4 +72,67 @@ public function itHasAnAttributeThatReturnsNearestDueDateOrNull()
...
+        ]);
+
+        // In Family3, sequence should be 1
+        $this->seeInDatabase('families', [
+            'id' => $family3->id,
+            'initial_centre_id' => $centre2->id,
+            'centre_sequence' => 1,
+        ]);
+    }
+
```
Comment (katjam) [inline tests/Unit/Models/FamilyModelTest.php:120]: Nice - though Technically you should test all the possible outcomes - including when you don't have a valid centre and when you find family with same centre - but that family has no centre_sequence. And when you give it a family that already has an RVID.

I'd be happy with those cases just added as todo.

**Review by katjam — APPROVED** [review]
Yeah - pretty much ready to go when you want. Just look over my comments. :open_umbrella:

File: app/Family.php (hunk 3 above)
Comment (charlesstrange2) [reply app/Family.php:226]: Nope. not yet.

**Pull request merged**